### PR TITLE
Dropped / moved some of old testsuites

### PR DESCRIPTION
**Pull request by mchf**
providing it for review before it gets to big. However, dropping other old testsuites is WIP.

### CODE REVIEW

File: src/include/network/routines.rb [hunk 1]
```diff
@@ -461,37 +461,27 @@ def DistinguishedName(name, hwdevice)
...
 
-      vendor = Ops.get_string(hwdevice, "sub_vendor", "")
-      dev = Ops.get_string(hwdevice, "sub_device", "")
+      vendor = hwdevice["sub_vendor"] || ""
+      dev = hwdevice["sub_device"] || ""
 
-      if vendor == "" || dev == ""
-        vendor = Ops.get_string(hwdevice, "vendor", "")
-        dev = Ops.get_string(hwdevice, "device", "")
+      if vendor.empty? || dev.empty?
```
Comment (jreidinger) [inline src/include/network/routines.rb]: it would be nice to have comment why sub_vendor have precedence before vendor

File: test/bond_test.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        expect(
+          LanItems
+            .GetBondableInterfaces(LanItems.GetCurrentName)
+            .map { |i| LanItems.GetDeviceName(i) }
+        ).to match_array expected_bondable
+      end
+    end
+
+    describe "#GetBondSlaves" do
+      it "returns list of slaves if bond device has some" do
```
Comment (jreidinger) [inline test/bond_test.rb]: maybe better to use "when" instead of "if"

File: test/bond_test.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        expect(LanItems.GetBondSlaves("bond0")).to match_array ["eth1", "eth2"]
+      end
+
+      it "returns empty list if bond device doesn't have slaves assigned" do
```
Comment (jreidinger) [inline test/bond_test.rb]: same here

File: test/routines_test.rb [hunk 4]
```diff
@@ -39,3 +39,45 @@ def initialize
...
+    expect(routines.ValidNicName("eth:0")).to be true
+  end
+
+  it "fails in case of long name" do
+    expect(routines.ValidNicName("0123456789012345")). to be false
+  end
+
+  it "fails when it contains invalid character" do
+    expect(routines.ValidNicName("eth0?")).to be false
+  end
```
Comment (jreidinger) [inline test/routines_test.rb]: missing test case that it fails also for empty device.

File: test/routines_test.rb [hunk 5]
```diff
@@ -39,3 +39,45 @@ def initialize
...
+    expect(routines.DeviceName({})).to be_empty
+  end
+
+  it "returns description containing model name when known" do
+    expect(routines.DeviceName(hwinfo.merge("model" => "hw_model"))).to eql "hw_model"
+  end
+
+  it "returns description build from vendor and device details when model name is uknown" do
+    expect(routines.DeviceName(hwinfo)).to eql "hw_vendor hw_device"
+  end
```
Comment (jreidinger) [inline test/routines_test.rb]: missing test case when only one of sub_ is defined, so vendor and device is used.

Comment (mchf) [inline test/routines_test.rb]: I've somehow expected comments like this. But I had to try to pass with only converting old testsuites ;-) ok, ok will extend it too ;-)

File: src/include/network/routines.rb [hunk 6]
```diff
@@ -461,37 +299,27 @@ def DistinguishedName(name, hwdevice)
...
+      "#{vendor} #{dev}".strip
     end
 
     # Validates given name for use as a nic name in sysconfig. See bnc#784952
     def ValidNicName(name)
       # 16 is the kernel limit on interface name size (IFNAMSIZ)
-      return false if !Builtins.regexpmatch(name, "^[[:alnum:]._:-]{1,15}$")
-
-      true
+      !(name =~ /^[[:alnum:]._:-]{1,15}$/).nil?
```
Comment (mvidner) [inline src/include/network/routines.rb]: `^` and `$` match LINES, not the entire string, in [Ruby regexes](http://ruby-doc.org/core-2.0.0/Regexp.html#class-Regexp-label-Anchors), so `"eth0\n" `would match. Use `\A` and `\z`.

Comment (kobliha) [inline src/include/network/routines.rb]: This is ideally covered by a unit test

**Comment by mvidner** [comment]
Thanks! Just one thing about Regexp, otherwise LGTM so far.

File: src/include/network/routines.rb (hunk 6 above)
Comment (mchf) [inline src/include/network/routines.rb]: Thanks. I've fixed the regexp and added testsuite for this case. These differences in regexp will never stop biting me.

**Comment by mchf** [comment]
Thanks for review. If travis succeeds I'll merge it. 

Subsequent conversions will follow later - this one is getting pretty big.

**Comment by jreidinger** [comment]
I propose to not merge it till we have SP1 branch. Reason is simple. Broken testsuite -> no submit request and it can block some P1 or shit stopper. Common problem I see is inappropriate arch mocking, which result e.g. in s390 build failing, so even jenkins do not discover it

**Comment by mchf** [comment]
well, yast2-networ has sp1 branch. however ok, i'll postpone merging. nothing urgent here

**Comment by mchf** [comment]
ok, branching is done - thanks to @ancorgs. Lets test it and merge ... ;-)
